### PR TITLE
[Feature] : Settings on folder level

### PR DESCRIPTION
**Pull request by bpoulaindev**
# Description

**Referencing issue #1310** 

Add settings at the folder level similarly to collection settings
- multiple settings tab can be opened
- only headers have been added in those settings
- headers var with the same name as one from the parent will override the parent's one in the request

image

### Contribution Checklist:

- [x] **The pull request only addresses one issue or adds one feature.**
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [x] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

### CODE REVIEW

**Comment by fulvio-diller** [comment]
It would be nice to have body and tests also..

**Comment by bpoulaindev** [comment]
@fulvio-diller Yeah i'll add those once headers are validated and officially stable

**Comment by kyleboe** [comment]
A thought from a design perspective: Should the concept of "Folders" be removed and should "Collections" be able to be nested? I'm not sure if that has already been discussed but it seems like that is the desired functionality as expressed in #1174 & #1310

Personally, my team would greatly benefit from being able to nest Collections to run a hierarchy of tests. So much so, that if we had that functionality, we would swap from Postman to Bruno immediately.

**Comment by tho-gru-38** [comment]
Hi @kyleboe,

I do not think that nesting collections is a good idea for several reasons:

1. How should bru.setNextRequest() implement a jump between nested collection?
2. What is the meaning of environments? Currently environments are stored at collection level. If a collection is nested there might exist environments too. To be able to run separately (standalone) environments must exist. On the other hand the values in the nested collection might be different from the values in the outer collection.
3. Currently it is not possible to move/copy a request from one collection to another collection.

To my understanding the concept of folder exist to create hierarchical tests. Therefore I want to be able to store scripts, tests ... at folder level.

Kind Regards
Thomas

File: packages/bruno-app/src/providers/ReduxStore/slices/collections/actions.js [hunk 1]
```diff
@@ -138,6 +139,61 @@ export const saveCollectionRoot = (collectionUid) => (dispatch, getState) => {
...
+};
+export const retrieveDirectoriesBetween = (pathname, parameter, filename) => {
+  const parameterIndex = pathname.indexOf(parameter);
+  const filenameIndex = pathname.indexOf(filename);
+  if (parameterIndex === -1 || filenameIndex === -1 || filenameIndex < parameterIndex) {
+    return [];
+  }
+  const directories = pathname
+    .substring(parameterIndex + parameter.length, filenameIndex)
+    .split('/')
```
Comment (Its-treason) [inline packages/bruno-app/src/providers/ReduxStore/slices/collections/actions.js:182]: `PATH_SEPERATOR` should be used here and 5 lines below for Windows compatible

File: packages/bruno-app/src/providers/ReduxStore/slices/collections/actions.js [hunk 2]
```diff
@@ -150,9 +206,26 @@ export const sendRequest = (item, collectionUid) => (dispatch, getState) => {
...
 
-    sendNetworkRequest(itemCopy, collection, environment, collectionCopy.collectionVariables)
+    const folderDatas = itemTree.reduce((acc, currentPath) => {
+      const folder = findItemInCollectionByPathname(collectionCopy, currentPath);
+      if (folder) {
+        acc = mergeRequests(acc, folder.root.request);
+      }
+      return acc;
+    }, {});
+    const mergeParams = mergeRequests(collectionCopy.root.request, folderDatas);
```
Comment (Its-treason) [inline packages/bruno-app/src/providers/ReduxStore/slices/collections/actions.js:259]: For new collection `collectionCopy.root` can be undefined resulting in an error here.

**Comment by helloanoop** [comment]
Merged to `feat/folder-level-settings` branch where I will be continue further work on this.

**Pull request merged**